### DISCUSSION TITLE
Rendering on demand in QtVSG

### DISCUSSION THREAD

**Discussion by dov** [vsg-dev/VulkanSceneGraph#1269] [General]
I'm working on an vsgQt based application which is static most of the time. However the vsg graph is still rerendered at a high framerate. How would I change the application so that the rendering is on demand only? Or similarly, that unless something changes, only the "last quad" is presented.

**Comment by robertosfield** [comment]
vsgQt::Viewer uses a Qt timer to drive the continues update, you can set the timer interval in ms to control the frame rate or toggle on/off the continuous update.

The vsgqtviewer enables control of both via the --interval <time> and --event-driven command line options.  From the example the command line parsing:

~~~ cpp
    bool continuousUpdate = !arguments.read({"--event-driven", "--ed"});
    auto interval = arguments.value<int>(8, "--interval");
~~~

And where these values are passed on when setting up the Viewer:

~~~ cpp
    if (interval >= 0) viewer->setInterval(interval);
    viewer->continuousUpdate = continuousUpdate;
~~~
